simpleviewinc/tap-events-force
# feat(filters): Hide waiting list filters

**Ticket**: [ZEN-491](https://jira.simpleviewtools.com/browse/ZEN-491)

## Context
* `Waiting list` and `on waiting list` labels should only be shown if one or more of the viewable sessions has **isWaitingListAvailable** is set to **true**
* Viewable sessions refer to the filtered sessions based on the attendees restriction


## Goal
* Only show the waiting list filter labels when at least one sessions has isWaitingListAvailable set to true

## Updates
* `src/actions/session/filters/applySessionFilters.js`
  * Added check to see if any sessions have a waiting list
* `src/actions/session/filters/setWaitingListFilters.js`
  * Added key to store, to save the waiting list active state
* `src/actions/session/mapSessionInterface.js`
  * Updated mapSessionInterface to call applySessionFilters
    * This way it's not calling setAgendaSessions twice on when initializing
* `src/components/modals/filter.js`
  * Updated filter modal to show waiting list labels based on it's state
* `src/hooks/sessions/useWaitingListActive.js`
  * Added hook to pull waiting list state from the store
* `src/components/sessions.js`
  * Added a small update to call `useStyle` instead of theme.get
* `src/mocks/eventsforce/bookingStates/waitingList.js`
  * Added test-data options to test no waiting list lable state

## Testing
**Test 1**
* Run the evf package => `keg evf pack run evf:zen-491-filter-modal-labels`
  * Open you browser to http://evf-zen-491-filter-modal-labels.local.kegdev.xyz/
    * Open the filters modal, and ensure `Waiting List` and `On Waiting List` are displayed
  * In the test-data editor, set the booking state to be `waitingList-single-noWaitingListLabel`, and hit apply
    * It's near the end of the list of booking state items
    * Open the filter modal and ensure the `Waiting List` and `On Waiting List` are **NOT** displayed
  * In the test-data editor, set the booking state to be `waitingList-group-noWaitingListLabel`, and hit apply
    * It's near the end of the list of booking state items
    * Open the filter modal and ensure the `Waiting List` and `On Waiting List` are **NOT** displayed

**Test 2**
* Run the consumer package => `keg consumer pack run consumer:zen-491-filter-modal-labels`
  * **NOTE**
    * The test-data have been updated to allow testing when no sessions has `isWaitingListAvailable` set `true`
    * You can **NOT** test other states in the test consumer
  * Open you browser to http://consumer-zen-491-filter-modal-labels.local.kegdev.xyz/
  * Open the filters modal, and ensure `Waiting List` and `On Waiting List` are **NOT** displayed
